Assets/Scripts/GridBlock.cs

```diff
@@ -10,23 +10,74 @@ public class GridBlock
 
     int y;
 
-    GridBlock(int xStart, int yStart, Block myBlock)
-    {
+    Block block;
+
+    Grid grid;
 
+    public GridBlock(int xStart, int yStart, Block myBlock)
+    {
+        x = xStart;
+        y = xStart;
+        block = myBlock;
     }
 
+    //Return the x index of the gridblock
     public int GetX()
     {
         return x;
     }
 
+    //Return the y index of the gridblock
     public int GetY()
     {
         return y;
     }
 
+    /*Enable gridblocks to move according to their directions.
+     Gridblock moves if it can fit into the next position.The 
+     moving process is not physical. Instead, it duplicates
+     all the tiles to the next position and erases the original
+     ones.*/
     public void Move(Enums.Direction direction)
```

Comment:
I recommend rewriting this method based on a new document I've written specifically for it: https://docs.google.com/document/d/1XDUkS8coz3WHC3MswyEs-dRWcZ8HEdpQcjDaAwLr52U/edit